server/events/policy_filter.go

```diff
@@ -58,18 +85,64 @@ func (p *ApprovedPolicyFilter) Filter(ctx context.Context, installationToken int
 	return filteredFailedPolicies, nil
 }
 
-func (p *ApprovedPolicyFilter) policyApproved(ctx context.Context, installationToken int64, approvedReviewers []string, failedPolicy valid.PolicySet) (bool, error) {
+func (p *ApprovedPolicyFilter) dismissStalePRReviews(ctx context.Context, installationToken int64, repo models.Repo, prNum int) error {
+	approvalReviews, err := p.prReviewFetcher.ListApprovalReviews(ctx, installationToken, repo, prNum)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch GH PR reviews")
+	}
+
+	latestCommitTimestamp, err := p.prLatestCommitFetcher.FetchLatestCommitTime(ctx, installationToken, repo, prNum)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch GH PR latest commit timestamp")
+	}
+
+	for _, approval := range approvalReviews {
+		// don't dismiss approvals after latest commit (check this first to avoid extra GH calls when possible)
+		if approval.GetSubmittedAt().After(latestCommitTimestamp) {
+			continue
+		}
+		isOwner, err := p.approverIsOwner(ctx, installationToken, approval)
+		if err != nil {
+			return errors.Wrap(err, "failed to validate approver is owner")
+		}
+		if isOwner {
+			err = p.prReviewDismisser.Dismiss(ctx, installationToken, repo, prNum, approval.GetID())
+			if err != nil {
+				return errors.Wrap(err, "failed to dismiss GH PR reviews")
+			}
+		}
+	}
+	return nil
+}
+
+func (p *ApprovedPolicyFilter) approverIsOwner(ctx context.Context, installationToken int64, approval *gh.PullRequestReview) (bool, error) {
+	if approval.GetUser() == nil {
+		return false, errors.New("failed to identify approver")
+	}
+	reviewers := []string{approval.GetUser().GetLogin()}
+	for _, policy := range p.policies {
+		isOwner, err := p.reviewersContainsPolicyOwner(ctx, installationToken, reviewers, policy)
+		if err != nil {
+			return false, errors.Wrap(err, "validating policy approval")
+		}
+		if isOwner {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (p *ApprovedPolicyFilter) reviewersContainsPolicyOwner(ctx context.Context, installationToken int64, reviewers []string, policy valid.PolicySet) (bool, error) {
 	// Check if any reviewer is an owner of the failed policy set
-	if p.ownersContainsPolicy(approvedReviewers, failedPolicy) {
+	if p.ownersContainsPolicy(reviewers, policy) {
 		return true, nil
 	}
-
 	// Cache miss, fetch and try again
-	err := p.getOwners(ctx, installationToken, failedPolicy)
+	err := p.fetchOwners(ctx, installationToken, policy)
 	if err != nil {
 		return false, errors.Wrap(err, "updating owners cache")
 	}
-	return p.ownersContainsPolicy(approvedReviewers, failedPolicy), nil
+	return p.ownersContainsPolicy(reviewers, policy), nil
 }
 
 func (p *ApprovedPolicyFilter) ownersContainsPolicy(approvedReviewers []string, failedPolicy valid.PolicySet) bool {
```

Comment:
it's not clear to me what the function does based on the naming, I'm assuming this checks whether any reviewers are policy owners.

Maybe `anyOwnerReviews` or something along those lines would make more sense.